modules/core/src/v2/coins/cspr.ts

```diff
@@ -105,9 +125,39 @@ export class Cspr extends BaseCoin {
   isValidAddress(address: string): boolean {
     throw new Error('Method not implemented.');
   }
-  signTransaction(paraCeloAccountLibms: any): Bluebird<SignedTransaction> {
-    throw new Error('Method not implemented.');
+
+  /**
+   * Assemble keychain and half-sign prebuilt transaction
+   *
+   * @param {SignTransactionOptions} params data required to rebuild and sign the transaction
+   * @param {TransactionPrebuild} params.txPrebuild prebuild object returned by platform
+   * @param {String} params.prv user prv used to sign the transaction
+   * @param {NodeCallback<SignedTransaction>} callback
+   * @returns Bluebird<SignedTransaction>
+   */
+  signTransaction(
+    params: SignTransactionOptions,
+    callback?: NodeCallback<SignedTransaction>
+  ): Bluebird<SignedTransaction> {
+    const self = this;
+    return co<SignedTransaction>(function*() {
+      const txBuilder = accountLib.getBuilder(self.getChain()).from(params.txPrebuild.txJson);
+      const key = params.prv;
+      txBuilder.sign({ key });
+
+      const transaction: any = yield txBuilder.build();
+      if (!transaction) {
+        throw new InvalidTransactionError('Error while trying to build transaction');
```

Comment:
Please create a new error type for this in `src/errors.ts` instead of throwing an error type from account-lib